advanced-achievements-plugin/src/test/java/com/hm/achievement/command/completer/CommandTabCompleterTest.java

```diff
@@ -0,0 +1,267 @@
+
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.ConsoleCommandSender;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.hm.achievement.command.executable.AbstractCommand;
+import com.hm.achievement.command.executable.BookCommand;
+import com.hm.achievement.command.executable.EasterEggCommand;
+import com.hm.achievement.command.executable.HelpCommand;
+import com.hm.mcshared.file.CommentedYamlConfiguration;
+
+/**
+ * Class for testing the command tab completer.
+ *
+ * @author Pyves
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class CommandTabCompleterTest {
+
+	@Mock
+	private CommentedYamlConfiguration mainConfig;
+
+	@Mock
+	private Command command;
+
+	@Mock
+	private CommandSender commandSender;
+
+	@Mock
+	private BookCommand bookCommand;
+
+	@Mock
+	private HelpCommand helpCommand;
+
+	@Mock
+	private EasterEggCommand easterEggCommand;
+
+	private CommandTabCompleter underTest;
+
+	@Before
+	public void setUp() {
+		when(command.getName()).thenReturn("aach");
+
+		when(mainConfig.getShallowKeys(any())).thenReturn(singleton("someSubcategory"));
+		when(mainConfig.getShallowKeys("Commands")).thenReturn(singleton("myCommand"));
+
+		Set<AbstractCommand> commands = new HashSet<>();
+		commands.add(bookCommand);
+		commands.add(easterEggCommand);
+		commands.add(helpCommand);
+		Map<String, String> achievementsAndDisplayNames = new HashMap<>();
+		achievementsAndDisplayNames.put("yourAch1", "Special Event Achievement!");
+		achievementsAndDisplayNames.put("yourAch2", "&2Coloured &rAchievement!");
+		achievementsAndDisplayNames.put("No Display Name Achievements!", "");
+		Set<String> disabledCategories = new HashSet<>();
+		disabledCategories.add("Crafts");
+		disabledCategories.add("Connections");
+		underTest = new CommandTabCompleter(mainConfig, achievementsAndDisplayNames, disabledCategories, commands, 13);
+		underTest.extractConfigurationParameters();
+	}
+
+	@Test
+	public void shouldReturnNullIfNotAachCommand() {
+		when(command.getName()).thenReturn("someothercommand");
+
+		List<String> completionResult = underTest.onTabComplete(commandSender, command, null, new String[0]);
+
+		assertNull(completionResult);
+	}
+
+	@Test
+	public void shouldReturnNullForThirdArgOfGiveCommand() {
+		String[] args = new String[] { "give", "yourAch1", "" };
+		List<String> completionResult = underTest.onTabComplete(commandSender, command, null, args);
+
+		assertNull(completionResult);
+	}
+
+	@Test
+	public void shouldReturnNullForThirdArgOfResetCommand() {
+		String[] args = new String[] { "reset", "Beds", "" };
+		List<String> completionResult = underTest.onTabComplete(commandSender, command, null, args);
+
+		assertNull(completionResult);
+	}
+
+	@Test
+	public void shouldReturnNullForThirdArgOfCheckCommand() {
```

Comment:
This test name does not describe that `shouldReturnPlayerList` method is being tested.

I think a comment would suffice for why we are expecting null - Another option is calling it _shouldReturnNullFor**Player**ArgOfCheckCommand_